administrator/components/com_associations/views/association/view.html.php

```diff
@@ -147,11 +147,30 @@ protected function addToolbar()
 		$input = JFactory::getApplication()->input;
 		$input->set('hidemainmenu', 1);
 
+		$user       = JFactory::getUser();
+		$userId     = $user->id;
+		$checkedOut = !($this->table->{$this->component->fields->checked_out} == 0 || $this->table->{$this->component->fields->checked_out} == $userId);
+		$component  = $input->get('acomponent', '', 'string');
+		$createdBy  = $this->component->fields->created_by;
+
 		JToolbarHelper::title(JText::_('COM_ASSOCIATIONS_HEADER_EDIT'), 'contract');
 
-		JToolbarHelper::apply('reference', 'COM_ASSOCIATIONS_SAVE_REFERENCE');
+		$canEdit = $user->authorise('core.edit', $component . "." . $this->associatedView . $this->referenceId);
```

Comment:
just reading the code i can see this is not correct ... did you even test?

if should be something like this

```php
$canEdit = $user->authorise('core.edit', $this->component->assetKey . '.' . $this->referenceId);
```